wallet/src/main/java/com/android/identity_credential/wallet/ui/qrscanner/ScanQrDialog.kt

```diff
@@ -0,0 +1,171 @@
+package com.android.identity_credential.wallet.ui.qrscanner
+
+import android.Manifest
+import android.view.ViewGroup
+import android.widget.LinearLayout
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.QrCode
+import androidx.compose.material3.AlertDialog
+import androidx.compose.material3.Button
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.compose.ui.window.DialogProperties
+import com.android.identity.util.Logger
+import com.android.identity_credential.wallet.R
+import com.budiyev.android.codescanner.CodeScanner
+import com.budiyev.android.codescanner.CodeScannerView
+import com.budiyev.android.codescanner.DecodeCallback
+import com.budiyev.android.codescanner.ErrorCallback
+import com.budiyev.android.codescanner.ScanMode
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.isGranted
+import com.google.accompanist.permissions.rememberPermissionState
+
+const val TAG = "ScanQrDialog"
+
+/**
+ * Composable Dialog that shows a [title], [description], Qr scanner/camera surface, and close button.
+ * Issues callbacks to [onClose] for dismiss requests (from dialog or close button). When a Qr code
+ * is successfully scanned, the callback [onScannedQrCode] is invoked with the decoded text.
+ *
+ * A [modifier] can be passed for setting the Dialog's dimensions
+ */
+@Composable
+fun ScanQrDialog(
+    title: String,
+    description: String,
+    onClose: () -> Unit,
+    onScannedQrCode: (String) -> Unit,
+    modifier: Modifier? = null
+) {
+    var dialogModier : Modifier = Modifier
```

Comment:
Both of these could be val instead of var, if the null check were moved up here... Seems like that'd be safer (it'd prevent accidental changes to these values).